`@InjectMock JsonWebToken jwt` fails with ContextNotActiveException

### Describe the bug

since `3.21.4` `@InjectMock JsonWebToken jwt` fails with ContextNotActiveException, it was running in `3.21.3`

```
org.junit.jupiter.api.extension.TestInstantiationException: Failed to create test instance
        at io.quarkus.test.junit.QuarkusTestExtension.initTestState(QuarkusTestExtension.java:767)
        at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:733)
        at java.base/java.util.Optional.orElseGet(Optional.java:364)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: jakarta.enterprise.context.ContextNotActiveException: RequestScoped context was not active when trying to obtain a bean instance for a client proxy of PRODUCER_METHOD bean [class=io.quarkus.smallrye.jwt.runtime.auth.JwtPrincipalProducer, id=BxDM8UQSQwnNG8ZE6VKFZ8wrsgg]
        - you can activate the request context for a specific method using the @ActivateRequestContext interceptor binding
```

### Expected behavior

should work imho as `@InjectMock` should somehow replace the real implementation bound to `@RequestScope`

### Actual behavior

fails with exception above

### How to Reproduce?

https://github.com/flyinfish/quarkus--inject-mock-jwt

### Output of `uname -a` or `ver`

Linux C70C7391 5.15.167.4-microsoft-standard-WSL2 #1 SMP Tue Nov 5 00:21:55 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk 21.0.7 2025-04-15 LTS OpenJDK Runtime Environment Zulu21.42+20-SA (build 21.0.7+6-LTS) OpenJDK 64-Bit Server VM Zulu21.42+20-SA (build 21.0.7+6-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.21.4

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.9 (8e8579a9e76f7d015ee5ec7bfcdc97d260186937) Maven home: /opt/apache-maven-3.9.9 Java version: 21.0.7, vendor: Azul Systems, Inc., runtime: /usr/lib/jvm/mobi-azuljdk21 Default locale: en, platform encoding: UTF-8 OS name: "linux", version: "5.15.167.4-microsoft-standard-wsl2", arch: "amd64", family: "unix"

### Additional information

_No response_

--- a/integration-tests/injectmock/src/main/java/io/quarkus/it/mockbean/RequestScopedFooFromProducer.java
+++ b/integration-tests/injectmock/src/main/java/io/quarkus/it/mockbean/RequestScopedFooFromProducer.java
@@ -1,0 +1,20 @@
+package io.quarkus.it.mockbean;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import jakarta.annotation.PostConstruct;
+
+public class RequestScopedFooFromProducer {
+
+    static final AtomicBoolean CONSTRUCTED = new AtomicBoolean();
+
+    public String ping() {
+        return "bar";
+    }
+
+    @PostConstruct
+    void init() {
+        CONSTRUCTED.set(true);
+    }
+
+}

--- a/integration-tests/injectmock/src/main/java/io/quarkus/it/mockbean/RequestScopedProducer.java
+++ b/integration-tests/injectmock/src/main/java/io/quarkus/it/mockbean/RequestScopedProducer.java
@@ -1,0 +1,17 @@
+package io.quarkus.it.mockbean;
+
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.enterprise.inject.Produces;
+
+import io.quarkus.arc.Unremovable;
+
+public class RequestScopedProducer {
+
+    @Produces
+    @RequestScoped
+    @Unremovable
+    public RequestScopedFooFromProducer produce() {
+        return new RequestScopedFooFromProducer();
+    }
+
+}

--- a/integration-tests/injectmock/src/test/java/io/quarkus/it/mockbean/RequestScopedFooMockTest.java
+++ b/integration-tests/injectmock/src/test/java/io/quarkus/it/mockbean/RequestScopedFooMockTest.java
@@ -15,11 +15,17 @@ class RequestScopedFooMockTest {
     @InjectMock
     RequestScopedFoo foo;
 
+    @InjectMock
+    RequestScopedFooFromProducer foo2;
+
     @Test
     void testMock() {
         when(foo.ping()).thenReturn("pong");
+        when(foo2.ping()).thenReturn("pong2");
         assertEquals("pong", foo.ping());
+        assertEquals("pong2", foo2.ping());
         assertFalse(RequestScopedFoo.CONSTRUCTED.get());
+        assertFalse(RequestScopedFooFromProducer.CONSTRUCTED.get());
     }
 
 }

--- a/test-framework/junit5-mockito/src/main/java/io/quarkus/test/junit/mockito/internal/MockitoMocksTracker.java
+++ b/test-framework/junit5-mockito/src/main/java/io/quarkus/test/junit/mockito/internal/MockitoMocksTracker.java
@@ -48,6 +48,17 @@ final class MockitoMocksTracker {
         return Optional.empty();
     }
 
-    record Mocked(Object mock, Object beanInstance) {
+    // don't use a Record because we don't want the auto-generated methods which delegate to the components
+    // see https://github.com/quarkusio/quarkus/issues/47739
+    @SuppressWarnings("ClassCanBeRecord")
+    final static class Mocked {
+
+        final Object mock;
+        final Object beanInstance;
+
+        Mocked(Object mock, Object beanInstance) {
+            this.mock = mock;
+            this.beanInstance = beanInstance;
+        }
     }
 }

--- a/test-framework/junit5-mockito/src/main/java/io/quarkus/test/junit/mockito/internal/SetMockitoMockAsBeanMockCallback.java
+++ b/test-framework/junit5-mockito/src/main/java/io/quarkus/test/junit/mockito/internal/SetMockitoMockAsBeanMockCallback.java
@@ -18,9 +18,9 @@ public class SetMockitoMockAsBeanMockCallback implements QuarkusTestBeforeEachCa
 
     private void installMock(MockitoMocksTracker.Mocked mocked) {
         try {
-            QuarkusMock.installMockForInstance(mocked.mock(), mocked.beanInstance());
+            QuarkusMock.installMockForInstance(mocked.mock, mocked.beanInstance);
         } catch (Exception e) {
-            throw new RuntimeException(mocked.beanInstance()
+            throw new RuntimeException(mocked.beanInstance
                     + " is not a normal scoped CDI bean, make sure the bean is a normal scope like @ApplicationScoped or @RequestScoped."
                     + " Alternatively you can use '@MockitoConfig(convertScopes = true)' in addition to '@InjectMock' if you would like"
                     + " Quarkus to automatically make that conversion (you should only use this if you understand the implications).");

